database/models/sensors.js

```diff
@@ -108,6 +182,36 @@ module.exports = {
                     else
                         resolve(result.rows);
                 });
+            })
+            // then get the corresponding inputs for each sensor
+            .then(function(sensors){
+
+                var sensorPs = sensors.map(function(sensor){
+                    var query = outputTable
+                    .select('*')
+                    .where(outputTable.sensor_id.equals(sensor.id))
```

Comment:
same here. do `.where(outputTable.sensor_id.in(sensors.map(function(s){ return s.id })))`